### PR TITLE
Prodiff teacher forcing inference

### PR DESCRIPTION
**Pull request by KLohithSaradhi**
## What?

Added scripts to handle "use_teacher_forcing" argument appropriately and call the inference function from SpectrogramDenoiser class.

## Why?

The "use_teacher_forcing" parameter in decode.yaml was not handled properly by the inference script for ProDiff.

### CODE REVIEW

File: espnet2/tts/prodiff/prodiff.py [hunk 1]
```diff
@@ -661,9 +672,13 @@ def _forward(
             h_masks = None
 
         if self.decoder_type == "diffusion":
-            before_outs = self.decoder(
-                hs, ys, h_masks, is_inference
-            )  # (B, T_feats, odim)
+            if not is_inference:
```
Comment (Fhrozen) [inline espnet2/tts/prodiff/prodiff.py:675]: Is this change really necessary?
The code at line https://github.com/espnet/espnet/blob/513318e31e46b0a98eabe0bab8948d3cec71db25/espnet2/tts/prodiff/denoiser.py#L237
performs only inference, so it seems unnecessary to me. 
Could you please check and see if there is any difference between your proposed code and the original? If not, please revert it, to approve your PR.

**Comment by Fhrozen** [comment]
@sw005320 
LGTM, except for one part. If fixed, then I think that it can be merged.

File: espnet2/tts/prodiff/prodiff.py (hunk 1 above)
Comment (KLohithSaradhi) [reply espnet2/tts/prodiff/prodiff.py:675]: Oh, my bad. In my trials to get the teacher forcing working ASAP, I just removed the "is_inference" parameter being passed because teacher forcing was used using training. Later when I was trying to write a clean solution, somehow  I skipped this line.

I'll fix this and raise another pull request.

File: espnet2/tts/prodiff/prodiff.py [hunk 2]
```diff
@@ -664,6 +675,8 @@ def _forward(
             before_outs = self.decoder(
                 hs, ys, h_masks, is_inference
             )  # (B, T_feats, odim)
+
+
```
Comment (Fhrozen) [inline espnet2/tts/prodiff/prodiff.py:679]: You added two black spaces. Please remove them.

File: espnet2/tts/prodiff/prodiff.py [hunk 3]
```diff
@@ -591,7 +591,7 @@ def _forward(
         lids: Optional[torch.Tensor] = None,
         is_inference: bool = False,
         alpha: float = 1.0,
-        use_teacher_forcing = False,        # allows proper teacher forcing while inferencing
+        use_teacher_forcing=False,  # allows proper teacher forcing while inferencing
```
Comment (Fhrozen) [inline espnet2/tts/prodiff/prodiff.py:594]: `use_teacher_forcing: bool = False`

**Comment by Fhrozen** [comment]
@sw005320
LGTM,

**Comment by sw005320** [comment]
Thanks!

@Fhrozen, do you know what happens in CI?

**Comment by Fhrozen** [comment]
Cannot install Espnet due to an error building `nnmnkwii`. Seems that `nnmnkwii` is using numpy.int instead of in:
`def fancy_plus_equals(cnp.ndarray[cnp.int_t, ndim=1] target_index_seq`. Needs to be updated for new numpy versions.

**Comment by sw005320** [comment]
> Cannot install Espnet due to an error building `nnmnkwii`. Seems that `nnmnkwii` is using numpy.int instead of in: `def fancy_plus_equals(cnp.ndarray[cnp.int_t, ndim=1] target_index_seq`. Needs to be updated for new numpy versions.

Thanks!
Can I ask you to finish it?

**Comment by Fhrozen** [comment]
@sw005320 it will better to ask @r9y9 to update the numpy environment in https://github.com/r9y9/nnmnkwii (change cnp.int to int)

**Comment by KLohithSaradhi** [comment]
@Fhrozen @sw005320 I'll be happy to make anymore changes to the code for it to get approved, if needed.

**Comment by Fhrozen** [comment]
@KLohithSaradhi, Thank you for the support, but the current issue with CI is a compatibility issue with numpy in another package, which I do not have direct access to.

**Comment by sw005320** [comment]
> @sw005320 it will better to ask @r9y9 to update the numpy environment in https://github.com/r9y9/nnmnkwii (change cnp.int to int)

@r9y9, could you deal with this?

**Comment by r9y9** [comment]
should be fixed by https://github.com/r9y9/nnmnkwii/releases/tag/v0.1.3

**Comment by sw005320** [comment]
> should be fixed by https://github.com/r9y9/nnmnkwii/releases/tag/v0.1.3

Thx!
I'll try it.

**Comment by sw005320** [comment]
@KLohithSaradhi,
Can you fix the CI error? 
See https://github.com/espnet/espnet/actions/runs/9726732023/job/26845494662

**Comment by KLohithSaradhi** [comment]
Should I take a look into this too:
`File "/home/runner/work/espnet/espnet/tools/venv/lib/python3.10/site-packages/s3prl/upstream/data2vec/data2vec_model.py", line 18, in <module>
    from omegaconf import II`

**Comment by KLohithSaradhi** [comment]
@sw005320 I have fixed the line length issues, sir. Am I required to do anything else?

**Comment by KLohithSaradhi** [comment]
@Fhrozen @sw005320 Am I required to fix more issues?

**Comment by sw005320** [comment]
Hi @KLohithSaradhi,
The current issue comes from s3prl and we're figuring out how to solve this.
After it is fixed, I'll merge your PR.

**Comment by KLohithSaradhi** [comment]
@sw005320 @Fhrozen My branch is outdated, should I close this PR and open a new one?

**Comment by Fhrozen** [comment]
You just need to merge with the last update from the main branch.

**Comment by sw005320** [comment]
Thanks @KLohithSaradhi and @Fhrozen!
